pkg/providers/aws/credentials_sts_manager.go

```diff
@@ -36,10 +36,8 @@ func NewSTSCredentialManager(client client.Client) *STSCredentialManager {
 
 //ReconcileProviderCredentials Ensure the credentials the AWS provider requires are available
 func (m *STSCredentialManager) ReconcileProviderCredentials(ctx context.Context, _ string) (*Credentials, error) {
-	secret, err := getSTSCredentialsSecret(ctx, m.Client, m.OperatorNamespace)
-	if err != nil {
-		return nil, errorUtil.Wrapf(err, "failed to get aws sts credentials secret %s", defaultSTSCredentialSecretName)
-	}
+	// if execution has reached this point, the secret exists - it was validated in NewCredentialManager()
+	secret, _ := getSTSCredentialsSecret(ctx, m.Client, m.OperatorNamespace)
```

Comment:
I would prefer keeping the error check. While I understand this is an edge case, technically it can happen. For example, secret was present during `NewCredentialManager()`, but was unable to get the secret for some reason / got deleted just before this function call, it would cause a nil pointer here and cause the operator to crash :thinking: